src/js/lib/cachetasks.js

```diff
@@ -191,6 +191,27 @@
       }, this);
     },
     /**
+     * Update the response associated with a request.
+     *
+     * @param  {Request} request   The request to check.
+     * @param  {String} cacheName The name of the cache to use.
+     * @param  {Object} options    options
+     *                              - force: true or false
+     * @return {Promise} Resolves once operations complete.
+     */
+    update(request, cacheName, options = {force: false}) {
+      return async.task(function*() {
+        let cache = yield MemoizedCaches.open(cacheName);
+        let response = yield cache.match(request);
+        let shouldUpdate = !response || options.force || this.isStale(response);
+        if (shouldUpdate) {
+          response = yield fetch(request);
```

Comment:
> In this case, I think the prudent option would be to return the cached request even if it's stale, to ensure a working application.

Yeah, I thought of doing that. My rationale was to allow the fetch to throw and allow the calling code to deal with it. Add add the recovery code into the method. 

> This is a scenario which needs to be defined and tested.

It's going to be hard to test, because we need to somehow force the Karma's test server to somehow produce a connection error :( Alternatively, we would need to make Gecko go-offline, but that would also break Karma as it relies on a socket connection :crying_cat_face: :sob: